### ISSUE TITLE
Deleting a site results in a signout error

### ISSUE THREAD

**Issue by mpope-tableau** [tableau/server-client-python#60]
I'm deleting a site using code like this:

```
tableau_auth = TSC.TableauAuth(args.username, password, site=args.contentUrl)
with server.auth.sign_in(tableau_auth):
    server.sites.delete(site_id)
    print("\nSite deleted.")
```
Note that in order to delete a site, I have to sign into that site by including the **contentUrl**. (The **site_id** value was fetched earlier in the program from which I take this snippet.) This code works--the site is deleted. 

_However_, at the termination of the **with** block, the code implicitly attempts to sign out from the site. But the site doesn't exist any more, so the signout fails and throws all sorts of errors.

**Comment by LGraber** [comment]
we will have to take a look at this. The server should allow you to signout after you delete the site. If you need to work around this ... don't use the "with" statement (and thus don't call signout). Thanks. Let's keep this open to track.

**Comment by t8y8** [comment]
This is one edge case that results from our REST API design, it would repro without the library.

After you delete the site your session is invalid (because sessions are tied to a site), and so every subsequent call will throw a 401.

We've known about it, and I swear documented it before.

**Comment by RussTheAerialist** [comment]
Even still, we should handle this situation better than throwing an exception out of the context handler.

**Comment by t8y8** [comment]
Yeah it's not a great experience... calls throw a generic 401.

I haven't looked at the __exit__ but we could try and catch it there... except it could be throwing a 401 for a valid reason. Actually, if you call sign out in the with block I wonder if this repros.

Or, the __exit__ could swallow all 401s because it's assumed you're signing out and don't care?

**Comment by LGraber** [comment]
I understand we may chose to "hide" this in the client code but this is a bug in our server code. I would agree that all other actions on the site should fail, but signout should not fail just because I deleted the site. That call should succeed because there is no reason it can't. I do not want to hide this in our client code until we have investigated fixing this behavior in our server. I feel confident that we can fix this in our server and do not want any change to our library yet.

**Comment by t8y8** [comment]
How? The site gets deleted the session is invalid.

The 401 is real -- what would you have Server do instead?

**Comment by LGraber** [comment]
Let's talk about this offline first.

**Comment by t8y8** [comment]
@LGraber had an idea on how to make sign_out a no-op without missing valid exceptions.

@mpope-tableau can I swing by and try a monkey-patch in your code?

**Comment by LGraber** [comment]
@mpope-tableau Note that you could have gotten the site_id off of the server object after you signed in. We expose the id of the site that you are logged into.

**Comment by t8y8** [comment]
Closed by #61
